tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/OrderingPartyControllerTests.cs

```diff
@@ -113,78 +121,85 @@ public static void Constructor_VerifyGuardClauses()
             [Frozen] CallOffId callOffId,
             Order order,
             OrderingPartyModel model,
             OrderingPartyController controller)
         {
-            context.Order.Add(order);
-            await context.SaveChangesAsync();
-
             var originalAddress = order.OrderingParty.Address;
 
-            await controller.UpdateAsync(callOffId, model);
+            service.Setup(o => o.GetOrder(callOffId)).ReturnsAsync(order);
+            service.Setup(o => o.SetOrderingParty(order, It.IsAny<OrderingParty>(), It.IsAny<Contact>())).Callback(() =>
+            {
+                order.OrderingParty.Name = model.Name;
+                order.OrderingParty.OdsCode = model.OdsCode;
+            });
+            contactDetailsService.Setup(o => o.AddOrUpdateAddress(It.IsAny<Address>(), It.IsAny<AddressModel>()))
+                .Returns(order.OrderingParty.Address);
 
+            await controller.UpdateAsync(callOffId, model);
+            service.Verify(o => o.GetOrder(callOffId));
+            service.Verify(o => o.SetOrderingParty(order, It.IsAny<OrderingParty>(), It.IsAny<Contact>()));
             contactDetailsService.Verify(s => s.AddOrUpdateAddress(
                 It.Is<Address>(a => a == originalAddress),
                 It.Is<AddressModel>(a => a == model.Address)));
+
+            order.OrderingParty.Address.Should().BeEquivalentTo(originalAddress);
         }
 
         [Test]
         [InMemoryDbAutoData]
         public static async Task UpdateAsync_InvokesAddOrUpdatePrimaryContact(
             [Frozen] Mock<IContactDetailsService> contactDetailsService,
-            [Frozen] ApplicationDbContext context,
+            [Frozen] Mock<IOrderingPartyService> service,
             [Frozen] CallOffId callOffId,
             Order order,
             OrderingPartyModel model,
             OrderingPartyController controller)
         {
-            context.Order.Add(order);
-            await context.SaveChangesAsync();
-
             var originalContact = order.OrderingPartyContact;
 
-            await controller.UpdateAsync(callOffId, model);
+            service.Setup(o => o.GetOrder(callOffId)).ReturnsAsync(order);
+            service.Setup(o => o.SetOrderingParty(order, It.IsAny<OrderingParty>(), It.IsAny<Contact>())).Callback(() =>
+            {
+                order.OrderingParty.Name = model.Name;
+                order.OrderingParty.OdsCode = model.OdsCode;
+            });
+            contactDetailsService.Setup(o => o.AddOrUpdatePrimaryContact(
+                    It.Is<Contact>(c => c == originalContact),
+                    It.Is<PrimaryContactModel>(c => c == model.PrimaryContact)))
+                .Returns(order.OrderingPartyContact);
 
+            await controller.UpdateAsync(callOffId, model);
+            service.Verify(o => o.GetOrder(callOffId));
+            service.Verify(o => o.SetOrderingParty(order, It.IsAny<OrderingParty>(), It.IsAny<Contact>()));
```

Comment:
I don't think we need these additional `Verify` tests here as we're testing that `AddOrUpdatePrimaryContact` is invoked here.

Reply:
Removed `service.Verify()`